controllers/nutanixcluster_controller.go

```diff
@@ -189,17 +189,32 @@ func (r *NutanixClusterReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		return reconcile.Result{}, err
 	}
 
-	v3Client, err := getPrismCentralClientForCluster(ctx, cluster, r.SecretInformer, r.ConfigMapInformer)
+	v3Client, err := getPrismCentralV3ClientForCluster(ctx, cluster, r.SecretInformer, r.ConfigMapInformer)
 	if err != nil {
 		log.Error(err, "error occurred while fetching prism central client")
 		return reconcile.Result{}, err
 	}
 
 	rctx := &nctx.ClusterContext{
-		Context:        ctx,
-		Cluster:        capiCluster,
-		NutanixCluster: cluster,
-		NutanixClient:  v3Client,
+		Context:         ctx,
+		Cluster:         capiCluster,
+		NutanixCluster:  cluster,
+		NutanixClientV3: v3Client,
+	}
+
+	createV4Client, err := isPrismCentralV4Compatible(ctx, v3Client)
+	if err != nil {
+		log.Error(err, "error occurred while checking environment compatibility for Prism Central v4 APIs")
```

Comment:
We intentionally log the error but don't return?

Reply:
we should atleast clear the variable here 
<pre>
rctx.NutanixClientV4 = nil
</pre>

there is no side effect if we dont return error. it just means it will not call v4apis as we check for rctx.NutanixClientV4 to be nil or not everywhere

Reply:
@deepakm-ntnx the variable assignation happens right after this. Until then, the variable is nil.